File: Tests/Parsing/MessageFolderScrapingTests.m
//  MessageFolderScrapingTests.m
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

#import "AwfulParsingTestCase.h"
#import "AwfulMessageFolderScraper.h"

@interface MessageFolderScrapingTests : AwfulParsingTestCase

@end

@implementation MessageFolderScrapingTests

+ (Class)scraperClass
{
    return [AwfulMessageFolderScraper class];
}

- (void)testInbox
{
    NSArray *messages = [self scrapeFixtureNamed:@"private-list"];
    XCTAssertEqual(messages.count, (NSUInteger)4);
    NSArray *allMessages = [AwfulPrivateMessage fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(messages.count, allMessages.count);
    NSArray *allUsers = [AwfulUser fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(allUsers.count, (NSUInteger)3);
    
    AwfulPrivateMessage *tagged = [AwfulPrivateMessage fetchArbitraryInManagedObjectContext:self.managedObjectContext
                                                                    matchingPredicateFormat:@"from.username = 'CamH'"];
    XCTAssertEqualObjects(tagged.messageID, @"4549686");
    XCTAssertEqualObjects(tagged.subject, @"Re: Awful app etc.");
    XCTAssertEqual(tagged.sentDate.timeIntervalSince1970, 1348778940.);
    XCTAssertEqualObjects(tagged.threadTag.imageName, @"sex");
    XCTAssertTrue(tagged.replied);
    XCTAssertTrue(tagged.seen);
    XCTAssertFalse(tagged.forwarded);
}

@end

File: Tests/Scraping/MessageFolderScrapingTests.swift
//  MessageFolderScrapingTests.swift
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

import XCTest
import Awful

class MessageFolderScrapingTests: ScrapingTestCase {
    
    override class func scraperClass() -> AnyClass {
        return PrivateMessageFolderScraper.self
    }

    func testInbox() {
        let scraper = scrapeFixtureNamed("private-list") as PrivateMessageFolderScraper
        let messages = scraper.messages
        XCTAssertTrue(messages.count == 4)
        XCTAssertEqual(messages.count, PrivateMessage.numberOfObjectsInManagedObjectContext(managedObjectContext))
        XCTAssertTrue(User.numberOfObjectsInManagedObjectContext(managedObjectContext) == 3)
        
        let tagged = PrivateMessage.fetchArbitraryInManagedObjectContext(managedObjectContext, matchingPredicate: NSPredicate(format: "from.username = 'CamH'"))
        XCTAssertEqual(tagged.messageID, "4549686")
        XCTAssertEqual(tagged.subject!, "Re: Awful app etc.")
        XCTAssertEqual(tagged.sentDate!.timeIntervalSince1970, 1348778940)
        XCTAssertEqual(tagged.threadTag!.imageName!, "sex")
        XCTAssertTrue(tagged.replied)
        XCTAssertTrue(tagged.seen)
        XCTAssertFalse(tagged.forwarded)
    }
}
